File: smartlist/src/main/java/com/smartlist/enums/TimeoutConfig.java
package com.smartlist.enums;

import java.time.LocalDateTime;
import java.time.temporal.ChronoUnit;

public enum TimeoutConfig {

    EMAIL_INVITE_LINK(2 * 24 * 60 * 60 * 1000), //2 days
    EMAIL_VERIFY_LINK(2 * 24 * 60 * 60 * 1000), //2 days
    FORGOT_PASSWORD_LINK(2 * 24 * 60 * 60 * 1000), //2 days
    ASSESSMENT_DURATION(5 * 24 * 60 * 60 * 1000),
    ACCESS_TOKEN_DURATION(60 * 60 * 24),
    REFRESH_TOKEN_DURATION(60 * 60 * 24 * 7);

    private final Integer value;

    private TimeoutConfig(final Integer value){
        this.value = value;
    }

    public static boolean isExpired(final LocalDateTime dateTime, final TimeoutConfig timeout) {
        return dateTime.plus(timeout.getValue(), ChronoUnit.MILLIS).isBefore(LocalDateTime.now());
    }

    public Integer getValue() {
        return this.value;
    }
}

File: javascript/smartlist/smartlist-api-mongodb-project/src/config/timeout.config.ts
export class TimeoutConfig {
  static EMAIL_INVITE_LINK: number = 2 * 24 * 60 * 60 * 1000;
  static EMAIL_VERIFY_LINK: number = 2 * 24 * 60 * 60 * 1000;
  static FORGOT_PASSWORD_LINK: number = 2 * 24 * 60 * 60 * 1000;
  static ASSESSMENT_DURATION = 12 * 24 * 60 * 60 * 1000;
  static ACCESS_TOKEN_DURATION = 60 * 60 * 24;
  static REFRESH_TOKEN_DURATION = 60 * 60 * 24 * 7;

  static isExpired(date: Date, timeoutConfig: number): boolean {
    return date.getTime() + timeoutConfig < new Date().getTime();
  }
}
